ArduPlane/GCS_Mavlink.cpp

```diff
@@ -650,15 +650,14 @@ bool GCS_MAVLINK_Plane::handle_guided_request(AP_Mission::Mission_Command &cmd)
         // only accept position updates when in GUIDED mode
         return false;
     }
-    plane.guided_WP_loc = cmd.content.location;
-    
-    // add home alt if needed
-    if (plane.guided_WP_loc.relative_alt) {
-        plane.guided_WP_loc.alt += plane.home.alt;
-        plane.guided_WP_loc.relative_alt = 0;
+    Location tmp = cmd.content.location;
+    if (!tmp.change_alt_frame(Location::AltFrame::ABSOLUTE)) {
```

Comment:
This loses the fact the wp was originally terrain-relative so we don't set terrain following flag appropriately.